lib/scraped_page.rb

```diff
@@ -1,5 +1,74 @@
+# frozen_string_literal: true
 require 'scraped_page/version'
 
-module ScrapedPage
-  # Your code goes here...
+require 'nokogiri'
+require 'open-uri'
+require 'field_serializer'
+require 'scraped_page_archive'
+
+# Abstract class which scrapers can extend to implement their functionality.
+class ScrapedPage
+  include FieldSerializer
+
+  class Response
+    attr_reader :status, :headers, :body
+
+    def initialize(body:, status: 200, headers: {})
+      @status = status
+      @headers = headers
+      @body = body
+    end
+  end
+
+  module Strategy
+    class LiveRequest
+      def response(url)
+        log "Fetching #{url}"
+        response = open(url)
+        {
+          status:  response.status.first.to_i,
+          headers: response.meta,
+          body:    response.read,
+        }
+      end
+
+      private
+
+      def log(message)
+        warn "[#{self.class}] #{message}"
```

Comment:
I think I'd prefer this to be turned off by default, considering the problems that Morph has with lots of console output. Perhaps even for the first version we could have it only output warnings if an ENV var is set whilst we work out what a better approach should be?